include/execution.hpp

```diff
@@ -838,50 +839,81 @@ namespace std::execution {
   }
 
   /////////////////////////////////////////////////////////////////////////////
-  // [execution.senders.consumer.start_detached]
-  // TODO: turn this into start_detached
-  inline namespace __submit {
+  // NOT TO SPEC: __submit
+  inline namespace __submit_ {
     namespace __impl {
-      template <class _Sender, class _ReceiverId>
-        struct __receiver {
-          using _Receiver = remove_cvref_t<_ReceiverId>;
-          struct __wrap {
-            __receiver* __this_;
-            // Forward all tag_invoke calls, including the receiver ops.
-            template <__decays_to<__wrap> _Self, class... _As, invocable<__member_t<_Self, _Receiver>, _As...> _Tag>
-            friend auto tag_invoke(_Tag tag, _Self&& __self, _As&&... __as)
-                noexcept(is_nothrow_invocable_v<_Tag, __member_t<_Self, _Receiver>, _As...>)
-                -> invoke_result_t<_Tag, __member_t<_Self, _Receiver>, _As...> {
-              // If we are about to complete the receiver contract, delete the state as cleanup:
-              struct _g_t {
-                __receiver* __rcvr_;
-                ~_g_t() { delete __rcvr_; }
-              } _g{__one_of<_Tag, set_value_t, set_error_t, set_done_t> ? __self.__this_ : nullptr};
-              return ((_Tag&&) tag)((__member_t<_Self, _Receiver>&&) __self.__this_->__rcvr_, (_As&&) __as...);
+      template <class _SenderId, class _ReceiverId>
+        struct __operation {
+          using _Sender = __t<_SenderId>;
+          using _Receiver = __t<_ReceiverId>;
+          struct __receiver {
+            __operation* __op_state_;
+            // Forward all the receiver ops, and delete the operation state.
+            template <__one_of<set_value_t, set_error_t, set_done_t> _Tag, class... _As>
+              requires invocable<_Tag, _Receiver, _As...>
+            friend auto tag_invoke(_Tag __tag, __receiver&& __self, _As&&... __as)
+                noexcept(is_nothrow_invocable_v<_Tag, _Receiver, _As...>)
+                -> invoke_result_t<_Tag, _Receiver, _As...> {
+              // Delete the state as cleanup:
+              unique_ptr<__operation> __g{__self.__op_state_};
+              return __tag((_Receiver&&) __self.__op_state_->__rcvr_, (_As&&) __as...);
```

Comment:
This implementation has a bug in that if the call to `set_value` throws then the operation-state is destroyed and a subsequent call to, say `set_error()` will access a dangling receiver/op-state.